src/plugins/data/public/ui/search_bar/create_search_bar.tsx

```diff
@@ -214,9 +218,8 @@ export function createSearchBar({
           isRefreshPaused={refreshInterval.pause}
           filters={filters}
           query={query}
-          isEnhancementsEnabled={isEnhancementsEnabled}
-          queryEnhancements={queryEnhancements}
           settings={settings}
+          containerRef={containerRef}
```

Comment:
So having Ref is due to that the QueryEditorUI component needs to measure the dimensions of the query editor container (queryEditorRect). This requires a reference to the DOM element to call getBoundingClientRect? or there is other cases that will cause DOM to unmount?